### PR TITLE
[DOC] set klass correctly for series and dataframe set_axis

### LINKED ISSUE #30881: DOC: missing docstring substitution for Series/DataFrame.set_axis()
https://dev.pandas.io/docs/reference/api/pandas.Series.set_axis.html
https://dev.pandas.io/docs/reference/api/pandas.DataFrame.set_axis.html

image

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [x] closes #30881 
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
Thanks @MarcoGorelli for the PR. will look soon.

**Comment by simonjayhawkins** [comment]
sorry @MarcoGorelli struggling to get doc build locally and now AFK for the best part of the day.

**Comment by MarcoGorelli** [comment]
@simonjayhawkins no hurry :)

**Comment by simonjayhawkins** [comment]
@MarcoGorelli should probably also change `See Also` to have %(klass)s substitution and use substitution for `An object of same type as caller` otherwise lgtm.

**Comment by MarcoGorelli** [comment]
@simonjayhawkins sure, done - thanks for your review!

**Review by simonjayhawkins — APPROVED** [review]
Thanks @MarcoGorelli for the updates.

I missed it the first time round, but now that the docstrings are independent, I think could also update axis to `axis : %(axes_single_arg)s, default 0` and re-work the docstrings a bit (separate PR OK)

**Review by simonjayhawkins — APPROVED** [review]

File: pandas/core/generic.py [hunk 1]
```diff
@@ -588,67 +588,23 @@ def set_axis(self, labels, axis=0, inplace=False):
         labels : list-like, Index
             The values for the new index.
 
-        axis : {0 or 'index', 1 or 'columns'}, default 0
-            The axis to update. The value 0 identifies the rows, and 1
-            identifies the columns.
+        axis : %(axes_single_arg)s, default 0
```
Comment (WillAyd) [inline pandas/core/generic.py]: Is this actually being substituted?

**Review by WillAyd — CHANGES_REQUESTED** [review]
Any chance you can post a screenshot of the built docs?

**Comment by MarcoGorelli** [comment]
Sure, here's the output of
`python make.py clean`
`python make.py --single pandas.DataFrame.set_axis`
`python make.py --single pandas.Series.set_axis`


image
image

File: pandas/core/generic.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/core/generic.py]: Looks like it, see below

**Review by jreback — APPROVED** [review]
lgtm. @WillAyd

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @MarcoGorelli

**Pull request merged**